packages/payment-processor/src/payment/erc20-transferable-receivable.ts

```diff
@@ -0,0 +1,250 @@
+import {
+  ContractTransaction,
+  Signer,
+  BigNumberish,
+  providers,
+  BigNumber,
+  constants,
+  ethers,
+} from 'ethers';
+
+import { Erc20PaymentNetwork } from '@requestnetwork/payment-detection';
+import { ERC20TransferableReceivable__factory } from '@requestnetwork/smart-contracts/types';
+import { ClientTypes, ExtensionTypes } from '@requestnetwork/types';
+
+import { ITransactionOverrides } from './transaction-overrides';
+import {
+  getAmountToPay,
+  getProxyAddress,
+  getProvider,
+  getSigner,
+  validateRequest,
+  getRequestPaymentValues,
+  validateMintERC20TransferableReceivable,
+} from './utils';
+import { IPreparedTransaction } from './prepared-transaction';
+
+import MultiFormat from '@requestnetwork/multi-format';
+
+// The ERC20 receivable smart contract ABI fragment
+const erc20TransferableReceivableContractAbiFragment = [
+  'function receivableTokenIdMapping(bytes32) public view returns (uint256)',
+];
+
+/**
+ * Gets the receivableTokenId from a ERC20TransferableReceivable contract given
+ * a paymentReference and paymentAddress of the request
+ * @param request
+ * @param signerOrProvider the Web3 provider, or signer. Defaults to window.ethereum.
+ */
+export async function getReceivableTokenIdForRequest(
+  request: ClientTypes.IRequestData,
+  signerOrProvider: providers.Provider | Signer,
+): Promise<BigNumber> {
+  // Setup the ERC20 proxy contract interface
+  const contract = new ethers.Contract(
+    getProxyAddress(
+      request,
+      Erc20PaymentNetwork.ERC20TransferableReceivablePaymentDetector.getDeploymentInformation,
```

Comment:
For some reason it isn't exposed at the top package level, I can only import it through Erc20PaymentNetwork first, which is what the other packages do... Will leave it like this since I don't want to mess up the package

Reply:
Please export `ERC20TransferableReceivablePaymentDetector` from `packages/payment-detection/src/index.ts`

Reply:
will do!